tensorflow/core/grappler/optimizers/generic_layout_optimizer_transposer.cc

```diff
@@ -538,10 +538,11 @@ bool Transposer::IsFaninPortDimsNIfConst(const utils::MutableNodeView& node,
     if (!tensor.FromProto(value_attr->tensor())) {
       return false;
     }
-    if (tensor.dims() != dims.size()) {
+    int dims_size = dims.size(); 
```

Comment:
```suggestion
    const int dims_size = dims.size(); 
```

Let's try to make these const if they don't change to enable a few compiler optimizations.